CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -432,12 +432,14 @@ public function assignProcessors() {
       }
       $this->_processors = array();
       foreach ($this->_paymentProcessors as $id => $processor) {
-        $this->_processors[$id] = ts($processor['name']);
-        if (!empty($processor['description'])) {
-          $this->_processors[$id] .= ' : ' . ts($processor['description']);
-        }
-        if ($processor['is_recur']) {
-          $this->_recurPaymentProcessors[$id] = $this->_processors[$id];
+        if ($processor['is_test'] == ($this->_mode == 'test')) {
```

Comment:
I don't particuarly like the way our schema stores payment processors - the fact that every payment processor has 2 ids, one for test and the other for live, is just a lot of trouble. I'm pretty sure we wouldn't ever want this array to contain both at the same time, so I mostly stand by this patch, the question is do we ever want the test mode processors or should we just always filter them out. IIRC another oddity of our pp system is that when you want to submit a payment in test mode you use the id of the *live* processor, so maybe we just always need those?